Correct text describing valid_time (#83)
tlswg/tls-subcerts: draft-ietf-tls-subcerts.md

### Validating a Delegated Credential

On receiving a delegated credential and a certificate chain, the peer validates
the certificate chain and matches the end-entity certificate to the peer's

> expected identity. Define `expiry time` as the delegation certificate's

Reviewer: ```suggestion
expected identity. Define expiry time as the delegation certificate's
```

> expected identity. Define expiry time as the delegation certificate's
> notBefore value plus DelegatedCredential.cred.valid_time. Then perform the
> following checks:

Reviewer 2: ```suggestion
expected identity. It then performs the following checks with expiry time set to the 
delegation certificate's notBefore value plus DelegatedCredential.cred.valid_time:
```

On receiving a delegated credential and a certificate chain, the peer
validates the certificate chain and matches the end-entity certificate to the

> peer's expected identity.  It also takes the following steps, with the expiry

Reviewer 3: How about something like this:
```
Define `expiry time` as the delegation certificate's notBefore value plus DelegatedCredential.cred.valid_time. Then perform the following checks:
```

Author: Thanks! I'll update to use that phrasing.

validates the certificate chain and matches the end-entity certificate to the
peer's expected identity.  It also takes the following steps:

1. Validate that DelegatedCredential.cred.valid_time is no more than 7 days.
1. Verify that the current time is within the validity interval of the credential.
   This is done by asserting that the current time is no more than the
   delegation certificate's notBefore value plus DelegatedCredential.cred.valid_time.
1. Verify that the delegated credential's remaining validity time is no more than the maximum validity

>    period. This is done by asserting that the current time is no more than the delegation

Reviewer 4: Why did this need rewording?

Author: The previous wording was
```
This is done by asserting that the current time is no more than the delegation
certificate's notBefore value plus DelegatedCredential.cred.valid_time plus
the maximum validity period.
```
and translates to `now() <= notBefore + valid_time + 7d`, which is different from `notBefore + valid_time <= now() + 7d` (the new wording).

Reviewer 4: Gotcha. I would consider introducing the expiry time explicitly.

Reviewer 3: No, I don't think it translates to the expression using "now()" that you have above. As currently described, you can pre-generate a bunch of subcerts, but with this change you cannot.

Reviewer 4: The recipient validates the subcert, not the generator, so you can still pregenerate, you just can't use them until they are valid. Am I missing something?

Reviewer 3: I mis-read the definition of "expiry time."  I proposed a wording change that should make it more clear.  Thanks!
